internal/bsr/encode.go

```diff
@@ -46,60 +73,58 @@ func NewChunkEncoder(ctx context.Context, w io.Writer, c Compression, e Encrypti
 
 // Encode serializes a Chunk and writes it with the encoder's writer.
 func (e ChunkEncoder) Encode(ctx context.Context, c Chunk) (int, error) {
+	encode := encodeCachePool.Get().(*encodeCache)
+	encode.Reset()
+	defer encodeCachePool.Put(encode)
+
 	data, err := c.MarshalData(ctx)
 	if err != nil {
 		return 0, err
 	}
 
-	var buf bytes.Buffer
 	var compressor io.WriteCloser
 	switch c.GetType() {
 	// Header should not be compressed since we need to read it prior to knowing
 	// what compression was used to check the compression bit.
 	// End should not be compressed since it has no data and compressing an empty
 	// byte slice just adds data in the form of the compression magic strings.
 	case ChunkHeader, ChunkEnd:
-		compressor = newNullCompressionWriter(&buf)
+		compressor = newNullCompressionWriter(encode.compress)
 	default:
 		switch e.compression {
 		case GzipCompression:
-			compressor = gzip.NewWriter(&buf)
+			compressor = gzip.NewWriter(encode.compress)
 		default:
-			compressor = newNullCompressionWriter(&buf)
+			compressor = newNullCompressionWriter(encode.compress)
 		}
 	}
 
 	if _, err := compressor.Write(data); err != nil {
 		return 0, err
 	}
-	err = compressor.Close()
-	if err != nil {
-		return 0, err
-	}
-	length := buf.Len()
-
-	t := c.GetTimestamp().marshal()
+	compressor.Close()
```

Comment:
Shouldn't we still care about the error this might throw?

Reply:
Good catch, I will fix this bug